### ISSUE TITLE
Should carthage outdated include incompatible dependencies?

### ISSUE THREAD

**Issue by dcaunt** [Carthage/Carthage#1260]
- `carthage version`: 0.16
- `xcodebuild -version`: Xcode 7.3 Build version 7D175
- Are you using `--no-build`? No
- Are you using `--no-use-binaries`? Yes
- Are you using `--use-submodules`? No

**Cartfile**

```
github "Quick/Quick" ~> 0.9.1
github "Quick/Nimble" ~> 3.2.0
```

**Carthage Output**

```
The following dependencies are outdated:
Quick "v0.9.1" -> "v0.9.2"
```

The Carthage `outdated` command introduced in #598 is great.

Do you think it should also listed dependencies which are newer, but not compatible with the version specified in the Cartfile? Given the above Cartfile, I'm not notified about the incompatible newer major version of Quick/Nimble which is 4.0.0, but chances are I'd be interested to know that it exists.

**Comment by mdiep** [comment]
I definitely think `outdated` should include that information somehow. The current implementation was a great first step, but I think there's a lot of room for improvement.

**Comment by jaybowong** [comment]
How's this going? 

The carthage folder is added to gitignore in our project, and in this way we can reduce the size of our git repo. We need to set the very specific versions in Cartfile to make sure every developer has the same code. It would be really great to know about new versions with `carthage outdated` as __expected__. It's especially useful when the number of dependencies is 10 or more.

**Comment by mdiep** [comment]
> How's this going?

No one is working on it AFAIK. Pull requests welcome. 😄 

> We need to set the very specific versions in Cartfile to make sure every developer has the same code

This shouldn't matter. The `Cartfile.resolved` always has specific versions. The developers need only run `carthage bootstrap`.

**Comment by ikesyo** [comment]
Resolved by #2254.

**Comment by dcaunt** [comment]
Thanks @iv-mexx ! ✨
